rust/xaynet-analytics/src/data_combination/data_combiner.rs

```diff
@@ -208,13 +208,13 @@ where
     // TODO: return an iterator instead of Vec: https://xainag.atlassian.net/browse/XN-1517
     fn get_events_single_route(
         &self,
```

Comment:
However, you need to specify the type as soon as you use it

```rust
DataCombiner::<MockAnalyticsEventRepo, MockScreenRouteRepo>::get_events_single_route(&home_route, all_events);
```

Not sure if this is better
@acrrd what do you think?

Reply:
You should be able to do `Self::get_events_single_route`. But you can also move it out of the `impl` block.

Reply:
i'll remove it and use `Self::method` within the `impl`, but yes, for tests it'll be more verbose, for this PR at least. 

Two considerations:

1) with the `AnalyticsController` component coming up in the next PR i was thinking of passing the references of events and routes into `DataCombiner::init_data_points`  instead of struct attributes passed with `new`. which would simplify the generic types a lot

2) basically then `DataCombiner` becomes a collection on static functions, so maybe that's a sign it needs some refactoring and a better design. Cause each `DataPoint` could have its own `DataCombiner` (trait?), since we're doing the same thing for each `DataPoint`: pass a `DataPointMetadata` and filtered events, and the package them in a `Vec<DataPoint>`. But yeh, the refactoring deserves its own PR, i would say

Reply:
Regarding point 2, you'll see in the next commit that it cause a chain reaction, cause if we remove it there, we can remove it in all other places. i think i still prefer for now to have `DataCombiner` to basically be a collection of static methods for now, instead of many module level functions, and later with the refactor break up everything and hopefully it'll make more sense